api/app/Http/Controllers/CourseController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Course;
+use App\Models\CourseUser;
+use App\Models\Department;
+use Illuminate\Http\Request;
+
+class CourseController extends Controller
+{
+    public function index()
+    {
+        $courses = Course::all();
+
+        return response()->json($courses);
+    }
+
+    public function store(Request $request)
+    {
+        $request->validate(
+            [
+                'course' => ['required'],
+                'department' => ['required'],
+                'user_ids' => ['required'],
+            ]
+        );
+        $course_id = Course::create(
+            [
+                'department_id' => Department::where('department', $request['department'])->get('id')->first()->id,
+                'course' => $request['course'],
+            ]
+        )->id;
+
+        $user_ids = array_filter(explode(',', $request['user_ids']));
+        foreach ($user_ids as $user_id) {
+            CourseUser::create(
+                [
+                    'user_id' => $user_id,
+                    'course_id' => $course_id,
+                ]
+            );
+        }
```

Comment:
instead using foreach, you can use upsert or insert
Ref: https://laravel.com/docs/9.x/eloquent#upserts